flatnav/DistanceInterface.h

```diff
@@ -8,7 +8,6 @@
 namespace flatnav {
 
 enum class METRIC_TYPE { EUCLIDEAN, INNER_PRODUCT };
-enum class DistanceMode { Asymmetric, Symmetric };
```

Comment:
I had introduced this enum earlier, but I just didn't like that I was templating distance classes that don't distinguish between symmetric and asymmetric distances. 

For PQ, 

* Asymmetric means you don't quantize the query vector. 
* Symmetric means you compare two vectors that are already quantized. 

Both are used in different places, but during search asymmetric is supposed to yield higher recall although it is more expensive to compute since you have to compute a distance table (a costly matrix) for each query. You can check out the implementation details in `ProductQuantization.h`.

Reply:
I'm gonna revisit this in a separate PR so that we have this fix first.